zipline/utils/events.py

```diff
@@ -436,23 +438,33 @@ def date_func(self, dt, env):
         raise NotImplementedError
 
     def calculate_start_and_end(self, dt, env):
-        next_trading_day = _coerce_datetime(
-            env.add_trading_days(
-                self.td_delta,
-                self.date_func(dt, env),
+        new_date = self.date_func(dt, env)
+        if not new_date:
+            return
+
+        try:
+            next_trading_day = _coerce_datetime(
+                env.add_trading_days(
+                    self.td_delta,
+                    new_date,
+                )
             )
-        )
+        except (NoFurtherDataError, TypeError):
+            return
 
         # If after applying the offset to the start/end day of the week, we get
         # day in a different week, skip this week and go on to the next
         while next_trading_day.isocalendar()[1] != dt.isocalendar()[1]:
             dt += datetime.timedelta(days=7)
-            next_trading_day = _coerce_datetime(
-                env.add_trading_days(
-                    self.td_delta,
-                    self.date_func(dt, env),
+            try:
+                next_trading_day = _coerce_datetime(
+                    env.add_trading_days(
+                        self.td_delta,
+                        self.date_func(dt, env),
+                    )
                 )
-            )
+            except (NoFurtherDataError, TypeError):
```

Comment:
When do we expect a `TypeError`?  Since that's such a generic error type, is there any better way to handle it that won't potentially mask something else raising a `TypeError`?